au3/libraries/lib-effects/EffectOutputTracks.cpp

```diff
@@ -41,6 +42,19 @@ EffectOutputTracks::EffectOutputTracks(
 
    for (auto aTrack : trackRange) {
       auto pTrack = aTrack->Duplicate();
+      if (auto aWaveTrack = dynamic_cast<WaveTrack*>(aTrack))
```

Comment:
this makes me think that there are other places where duplicated track should also duplicate clip ids. I'm not opposed to have this merged, but what do you think about introduce this as an optional behavior to `Duplicate`?

Reply:
You find that this PR solves this ID problem only in the context of effects but that preserving clip IDs might be desired in other situations, too?
Where `Duplicate()` is called (I changed the name to something silly and fixed compilation to make sure I haven't missed any):
* `EffectOutputTracks`: IDs must be copied
* `ProjectCloudExtension::OnSyncStarted()` : I think it doesn't matter?
* `PendingTracks`: outdated ;)
* Undo/Redo: at the moment new IDs are given, and it seems to work (I don't know why)
* In `WaveTrack::ClearAndPaste`: new IDs are given, which is correct
* `Au3Interaction::duplicateTracks`: new IDs are given, correct.

So for now at least it would seem that we're good?

Yes, in the future a situation may arise where we need to duplicate with equal IDs. I'm not sure how to change the `Duplicate` API, though, as this is a `Track` API, and other track types may not have clips (e.g. label tracks).

Reply:
> Undo/Redo: at the moment new IDs are given, and it seems to work (I don't know why)

Now I wonder why applying an effect breaks the project while undo doesn't. If I understand correctly, undo does the same thing as an effect does:  it creates a new track and replaces the original track, including all of its clips.

Reply:
All things considered, I'll move this to QA already: having this merged would unblock the other ticket I'm working on.

Reply:
@kryksyh https://github.com/audacity/audacity/pull/7880